server/admin_management/api/app/routers/router_guards.py

```diff
@@ -121,71 +115,140 @@ async def get_target_user_from_id(
+        acpid = request.path_params["access_control_privilege_id"]
+        LOGGER.debug("Dependency 'get_target_user_from_id': 'access_control_privilege_id' " +
+                     f"path param found: {acpid}.")
         access_control_privilege = access_control_privilege_service.get_acp_by_id(
-            request.path_params["access_control_privilege_id"]
-        )
-        return (
-            TargetUser.model_validate(access_control_privilege.user)
-            if access_control_privilege is not None
-            else None
+            acpid
         )
+        if access_control_privilege is not None:
+            return TargetUser.model_validate(access_control_privilege.user)
+        else:
+            error_msg = "Parameter 'access_control_privilege_id' {acpid} is missing or invalid."
+            utils.raise_http_exception(
+                error_code=ERROR_CODE_INVALID_REQUEST_PARAMETER,
+                error_msg=error_msg
+            )
     else:
         # from body - {user_name/user_type_code}, when grant admin access
-        try:
-            rbody = await request.json()
-            user = user_service.get_user_by_domain_and_name(
-                rbody["user_type_code"],
-                rbody["user_name"],
+        rbody = await request.json()
+        LOGGER.debug(f"Dependency 'get_target_user_from_id' request body {rbody}.")
+        rb_user_name = rbody["user_name"]
+        rb_user_type_code = rbody["user_type_code"]
+        # user_guid is searched from IDIM search and passed from frontend.
+        rb_user_guid = rbody["user_guid"]
+
+        user = user_service.get_user_by_domain_and_name(
+            rb_user_type_code,
+            rb_user_name,
+        )
+        if user is not None:
+            found_target_user = TargetUser.model_validate(user)
+            # old data might not have user_guid.
+            if found_target_user.user_guid is None:
+                found_target_user.user_guid = rb_user_guid
+            return found_target_user
+        # user not found, case of new user.
+        else:
+            target_new_user = TargetUser.model_validate({
+                "user_name": rb_user_name,
+                "user_type_code": rb_user_type_code,
+                "user_guid": rb_user_guid
+            })
+            return target_new_user
+
+
+async def target_user_bceid_search(
+    requester: Requester = Depends(get_current_requester),
+    target_user: TargetUser = Depends(get_target_user_from_id)
+) -> TargetUser:
+    """
+    BCeID user search for target_user.
+    :param target_user "Depends(get_target_user_from_id)": the initial
+        dependency with "get_target_user_from_id" has initial target_user
+        parsed from the request. It's business_guid will be updated to the
+        target_user after user is searched through IDIM Proxy.
+    """
+    if (
+        target_user.user_type_code == UserType.BCEID and
+        (
+            target_user.is_new_user() or
```

Comment:
I wonder why we need to check if this is a new user or not? I feel as long as the `target_user.business_guid` is None we should run the following

Reply:
Thanks, that's right, either way it is empty, I over looking.